### PR TITLE
Changed brodcast code to work with jip and to prevent usage of public variables.

### PR DESCRIPTION
**Pull request by Zakant**
When merged this pull request will do:
- Change the `Ares_fnc_BroadcastCode` to support JIP. 
- Remove the need of a public variable, because `remoteExec` can work with codeblocks as well (Should slightly increase performance and reduce bandwith, because variables doesn't need to be broadcasted to all clients)
- Added support for different execution modes (e.g. execute where object is local).

If you have any questions feel free to ask. If you don't like my changes, you have ofcourse not to merge them. 
Zakant

O.T: I dont really see any advantage using a dedicated function to raise a remoteExecution insteand of using `remoteExec` directly. Have i missed something?

### CODE REVIEW

**Comment by oOKexOo** [comment]
I'm careful when it comes to remote execution, since the introduction of the CfgRemoteExec class. I actually took the script from default Ares but replaced the BIS_fnc_MP, since it was obsoleted. As I know that Ares worked fine, I did not further thought about it. In a first step, we have to ensure that the BIS_fnc_call respective BIS_fnc_spawn does not get blocked by a server. Otherwise, it will lead to an almost complete break down of the mod, what I definitely want to avoid.

**Comment by Zakant** [comment]
Hey, okay, i can totally understand that problem, haven't thought about it earlier. But to be honest: By default, remote execution is enabled for all functions. So they `BIS_fnc_call` should be only blocked IF a serveradmin has enabled restrictions on his server by changing the `CfgRemoteExec` (and has hopfully an idea what he just did...). So we should not expect any problems as long as people dont mess with their servers and have no idea what they are doing. I'll check soon if their is a way we could detect if whitelisting is enabled and `BIS_fnc_call` is whitelisted. If not we could display an error message. 
Tl;Dr: Should only be problem if somebody has changed pro active. Also ACE would create errors as well, 'couse their using `remoteExec` a lot.
If you are still feeling uncomfortable i'll to do some research about remote execution and report back.

Greetings Zakant

**Comment by oOKexOo** [comment]
I totally forgot to tell you that I already reorganized scripts and compartmentalized the pbo. I still have to think about that change carefully. But even if I agree, I have to integrate the change by hand.

**Comment by oOKexOo** [comment]
I used `remoteExec` in several modules so far and no one complained yet. So I decided to introduce `remoteExec`. For this purpose I replaced `Ares_fnc_BroadcastCode` by  `remoteExec` everywhere such taht the function is obsolete now. Hence, this pull is no longer relevant. However, I introduced your suggested change: JIP option will be available for the "Execute Code" module.
